lms/services/lti.py

```diff
@@ -0,0 +1,86 @@
+"""
+LTI service.
+
+This is a place to put LTI-related methods where they can be called from
+anywhere in the code, they can have access to the request and context, and they
+can have state (e.g. to cache the results of computations).
+"""
+import pylti.common
+
+from lms.services import LTILaunchVerificationError
+from lms.services import NoConsumerKey
+from lms.services import ConsumerKeyError
+from lms.services import LTIOAuthError
+
+
+class LTIService:
+    """A collection of LTI-related methods."""
+
+    def __init__(self, _context, request):
+        self._request = request
+        self._called = False
+        self._exception = None
+
+    def verify_launch_request(self):
+        """
+        Raise if the current request isn't a valid LTI launch request.
+
+        :raise ~lms.services.LTILaunchVerificationError: if the request isn't a
```

Comment:
How come the dotted path of this exception is spelled out in full but the others are not?

Reply:
Fixes